src/ocl/dropoutocl.cpp

```diff
@@ -135,15 +135,15 @@ void DropoutDescriptor::InitPRNGState(Handle& handle,
     size_t states_num = prng_stateSizeInBytes / sizeof(prngStates);
     size_t wk_grp_num = std::min(size_t(MAX_PRNG_STATE / 256), (states_num + 255) / 256);
 
-    std::string network_config = "initprngs-" + std::to_string(states_num) + "x" +
-                                 std::to_string(sizeof(prngStates)) + "x" +
-                                 std::to_string(rng_mode) + "x" + std::to_string(prng_seed) + "x" +
-                                 std::to_string(wk_grp_num);
+    std::string network_config = "initprngs-" + std::to_string(sizeof(prngStates)) + "x" +
+                                 std::to_string(rng_mode) + "x" + std::to_string(wk_grp_num);
 
     auto&& kernels = handle.GetKernels(kernel_name, network_config);
     if(!kernels.empty())
     {
-        kernels.front()(prng_states);
+        kernels.front()(prng_states,
```

Comment:
Maybe `prng_seed` and `states_num` should be declared as `unsigned`? That would make calls single lines and remove the casts.

Reply:
Fixed (through a different way). Changing prng_seed to be `unsigned` would change the MIOpen C API (in which `miopenSetDropoutDescriptor` uses `unsigned long long`). I'd rather pass in an unsigned long long value, which I now do.

Reply:
I have missed that. You should probably check if it would fit in unsigned, unless we don't really care. I don't think that would cause problems other than seed collisions.